android-client-sdk/src/main/java/com/devcycle/sdk/android/api/DVCClient.kt

```diff
@@ -298,14 +299,29 @@ class DVCClient private constructor(
 
     private suspend fun fetchConfig(user: User) {
         val now = System.currentTimeMillis()
-        val result = request.getConfigJson(environmentKey, user)
+        val result = request.getConfigJson(environmentKey, user, enableEdgeDB)
         config = result
         observable.configUpdated(config)
         dvcSharedPrefs.save(config, DVCSharedPrefs.ConfigKey)
 
         this@DVCClient.user = user
         saveUser()
         addUserConfigResultToEventQueue(now, this@DVCClient.user, config!!)
+
+        if (checkIfEdgeDBEnabled(config!!, enableEdgeDB)) {
+            if (!user.isAnonymous) {
+                request.saveEntity(user)
+            }
+        }
+    }
+
+    private fun checkIfEdgeDBEnabled(config: BucketedUserConfig, enableEdgeDB: Boolean): Boolean {
+        if (config.project?.settings?.edgeDB?.enabled == true) {
+            return !!enableEdgeDB
```

Comment:
What is the intention here?

Reply:
What is the intention of the double negation in front of `enableEdgeDB`?

Reply:
ah yes, JS habits since its not a strongly typed language.. its a check for a more explicit truthy check